src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][0], gridBagConstraints);
+
+		blocks[0][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][1], gridBagConstraints);
+
+		blocks[0][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][2], gridBagConstraints);
+
+		blocks[1][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 0, 0);
+		add(blocks[1][0], gridBagConstraints);
+
+		blocks[1][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+		add(blocks[1][1], gridBagConstraints);
+
+		blocks[1][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 1);
+		add(blocks[1][2], gridBagConstraints);
+
+		blocks[2][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 1, 0);
+		add(blocks[2][0], gridBagConstraints);
+
+		blocks[2][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(blocks[2][1], gridBagConstraints);
+
+		blocks[2][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(blocks[2][2], gridBagConstraints);
+	}
+
+	public void populatePanel(GameSession gameSession, boolean isRefresh) {
+
+		this.gameSession = gameSession;
+		this.selectedCell = null;
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),

```

Comment:
Need more info on this. Maybe a pseudo code?

On Sun, Jul 12, 2015 at 10:09 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429573>:
>
> > +
> > +		blocks[2][2] = new BlockGUI();
> > +		gridBagConstraints = new java.awt.GridBagConstraints();
> > +		gridBagConstraints.gridx = 2;
> > +		gridBagConstraints.gridy = 2;
> > +		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
> > +//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
> > +		add(blocks[2][2], gridBagConstraints);
> > +	}
> > +
> > +	public void populatePanel(GameSession gameSession, boolean isRefresh) {
> > +
> > +		this.gameSession = gameSession;
> > +		this.selectedCell = null;
> > +
> > +		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),
>
> Prefer getting the gameBoard once and reusing the variable rather than
> querying getGameBoard 9 times (performance improvement)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429573>.
>